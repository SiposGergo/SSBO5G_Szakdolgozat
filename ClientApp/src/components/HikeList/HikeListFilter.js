import React from "react";
import { connect } from "react-redux";
import {
    setTextFilter, sortByName, sortByDate,
    setEndDate, setStartDate, setOldHikesVisibility, setSliderValues
}
    from "../../actions/HikeListActions";
import "react-dates/initialize";
import { DateRangePicker } from "react-dates";
import Slider, { Range } from 'rc-slider';
import Tooltip from 'rc-tooltip';


const rangeMarks = {
    0: { label: "0 km" },
    10: { label: "10 km" },
    20: { label: "20 km" },
    30: { label: "30 km" },
    40: { label: "40 km" },
    50: { label: "50 km" },
    60: { label: "60 km" },
    70: { label: "70 km" },
    80: { label: "80 km" },
    90: { label: "90 km" },
    100: { label: "100 km" }

}

const createSliderWithTooltip = Slider.createSliderWithTooltip;
const RangeSlider = createSliderWithTooltip(Range);


export class HikeListFilter extends React.Component {
    state = {
        calendarFocused: null
    };
    onDateChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }
    onFocusChanged = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortChange = (e) => {
        if (e.target.value === "date") {
            this.props.sortByDate();
        }
        else if (e.target.value === "name") {
            this.props.sortByName();
        }
    };

    handleTextBoxClick = (event) => {
        const target = event.target;
        this.props.setOldHikesVisibility(target.checked);
    }

    onSliderChange = (value) => {
        this.props.setSliderValues(value);
    }

    render() {
        return (
            <div>
                <input type="text" placeholder="Keresés" defaultValue={this.props.filters.text} onChange={this.onTextChange} />
                <label>Rendezés: </label>
                <select
                    defaultValue={this.props.filters.sortBy}
                    onChange={this.onSortChange}
                >
                    <option value="date">Dátum</option>
                    <option value="name">Név</option>
                </select>
                <label>
                    Már megrendezett túrák mutatása
                <input type="checkbox" id="oldHikes" name="oldHikes"
                        onChange={this.handleTextBoxClick} />
                </label>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    startDateId="asd"
                    endDate={this.props.filters.endDate}
                    endDateId="lol"
                    onDatesChange={this.onDateChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChanged}
                    numberOfMonths={1}
                    showClearDates={true}
                    isOutsideRange={() => false} />
                <div>
                    <RangeSlider
                        min={0}
                        max={100}
                        marks={rangeMarks}
                        defaultValue={[0, 100]}
                        onChange={this.onSliderChange}
                    />
                </div>
            </div>
        )
    }
}
const mapStateToProps = (state) => ({
    filters: state
});

const mapDispatchToProps = (dispatch, props) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByName: () => dispatch(sortByName()),
    setStartDate: (date) => dispatch(setStartDate(date)),
    setEndDate: (date) => dispatch(setEndDate(date)),
    setOldHikesVisibility: (bool) => dispatch(setOldHikesVisibility(bool)),
    setSliderValues: (value) => dispatch(setSliderValues(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(HikeListFilter);