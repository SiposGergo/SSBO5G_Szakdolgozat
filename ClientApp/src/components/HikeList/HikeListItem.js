import React from "react";
import moment from "moment";
import { NavLink } from "react-router-dom";

class HikeListItem extends React.Component {

    render() {
        const props = this.props;
        return (<div>
            <h1><NavLink to={"/hike/" + props.hike.id} exact={true}>{props.hike.name}</NavLink></h1>
            <a href={props.hike.website} target="_blank">A Túra oldala</a>
            <table>
                <tbody>
                    <tr>
                        <td>Időpont:</td>
                        <td>{new moment(props.hike.date).format('YYYY. MM. DD.')}</td>
                    </tr>
                    <tr>
                        <td>Távok: </td>
                        <td>{props.hike.courses.length == 0 ? "Nincsenek távok"
                            : props.hike.courses.map(course =>
                                course.distance / 1000 + " km, ")}
                        </td>
                    </tr>
                </tbody>
            </table>
        </div>)
    }
}

export default HikeListItem;