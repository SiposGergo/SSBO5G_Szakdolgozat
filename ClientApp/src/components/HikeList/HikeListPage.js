import React from "react";
import { connect } from "react-redux";
import getVisibleHikes from "../../selectors/HikeListSelector";
import HikeListItem from "./HikeListItem";
import HikeListFilter from "./HikeListFilter.js";

import { itemsFetchData, reset } from "../../actions/HikeListActions"

const API = "http://localhost:4242/Hike/all";

export class HikeListPage extends React.Component {
    componentDidMount() {
        this.props.fetchData(API);
    }

    componentWillUnmount() {
        this.props.reset();
    }

    render() {
        if (this.props.hasErrored) {
            return <p>Sorry! There was an error loading the items</p>;
        }

        if (this.props.isLoading) {
            return <p>Loadingâ€¦</p>;
        }

        return ( <div>
                <HikeListFilter />
                {
                    this.props.hikes.length === 0 ? (
                        <p>No Hikes</p>
                    ) : (
                            this.props.hikes.map((h) => {
                                return (<HikeListItem key={h.id} hike={h} />)
                            })
                        )
                }
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        hikes: getVisibleHikes(state.HikeListReducer),
        hasErrored: state.HikeListReducer.hasErrored,
        isLoading: state.HikeListReducer.isLoading
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchData: (url) => dispatch(itemsFetchData(url)),
        reset: () => dispatch(reset())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(HikeListPage);