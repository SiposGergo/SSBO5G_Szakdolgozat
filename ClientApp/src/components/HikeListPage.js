import React from "react";
import { connect } from "react-redux";
import getVisibleHikes from "../selectors/hikes.js";
import HikeListItem from "./HikeListItem";
import HikeListFilter from "./HikeListFilter.js";

import { itemsFetchData } from "../actions/filters"

const API = "http://localhost:4242/Hike/all";

class HikeListPage extends React.Component {
    componentDidMount() {
        this.props.fetchData(API);
    }

    render() {
        if (this.props.hasErrored) {
            return <p>Sorry! There was an error loading the items</p>;
        }

        if (this.props.isLoading) {
            return <p>Loadingâ€¦</p>;
        }

        return (
            <div>
                <HikeListFilter />
                {
                    this.props.hikes.length === 0 ? (
                        <p>No Hikes</p>
                    ) : (
                            this.props.hikes.map((h) => {
                                return (<HikeListItem key={h.id} hike={h} />)
                            })
                        )
                }
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        hikes: getVisibleHikes(state),
        hasErrored: state.hasErrored,
        isLoading: state.isLoading

    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchData: (url) => dispatch(itemsFetchData(url))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(HikeListPage);