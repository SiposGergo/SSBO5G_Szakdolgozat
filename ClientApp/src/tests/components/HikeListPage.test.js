import React from "react";
import { shallow } from "enzyme";
import { HikeListPage } from "../../components/HikeList/HikeListPage";
import { input } from "../fixtures/testHikes";


test('HikeListPage should render correctly', () => {
    const wrapper = shallow(<HikeListPage {...input} fetchData={() => { }} />);
    expect(wrapper).toMatchSnapshot();
});

test('HikeListPage should render correctly while loading hikes', () => {
    const wrapper = shallow(<HikeListPage {...input} isLoading={true} fetchData={() => { }} />);
    expect(wrapper).toMatchSnapshot();
});

test('HikeListPage should render correctly with errors', () => {
    const wrapper = shallow(<HikeListPage {...input} hasErrored={true} fetchData={() => { }} />);
    expect(wrapper).toMatchSnapshot();
});