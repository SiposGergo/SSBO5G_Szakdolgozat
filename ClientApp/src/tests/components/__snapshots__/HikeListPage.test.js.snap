// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HikeListPage should render correctly 1`] = `
<div>
  <Connect(HikeListFilter) />
  <HikeListItem
    hike={
      Object {
        "comments": null,
        "courses": Array [
          Object {
            "beginningOfStart": "2018-03-09T07:00:00",
            "checkPoints": null,
            "distance": 43320,
            "elevation": 1922,
            "endOfStart": "2018-03-09T08:00:00",
            "hikeId": 1,
            "id": 1,
            "limitTime": "11:00:00",
            "maxNumOfHikers": 100,
            "name": "Mátrahegy 40",
            "participations": null,
            "placeOfFinish": "Mátrafüred, Mátra Szakképző Iskola (Erdész u. 11.)",
            "placeOfStart": "Mátrafüred, Mátra Szakképző Iskola (Erdész u. 11.)",
            "price": 1000,
            "registerDeadline": "2018-03-08T00:00:00",
            "registrations": null,
          },
        ],
        "date": "2018-03-09T00:00:00",
        "description": "Mátrahegy telejsítménytúra",
        "id": 1,
        "name": "Mátrahegy",
        "organizer": null,
        "organizerId": 3,
        "staff": null,
        "website": "http://kekesturista.hu",
      }
    }
    key="1"
  />
  <HikeListItem
    hike={
      Object {
        "comments": null,
        "courses": Array [
          Object {
            "beginningOfStart": "2018-07-22T06:00:00",
            "checkPoints": null,
            "distance": 51400,
            "elevation": 1992,
            "endOfStart": "2018-07-22T08:00:00",
            "hikeId": 2,
            "id": 2,
            "limitTime": "12:00:00",
            "maxNumOfHikers": 60,
            "name": "Patai Mátra 50",
            "participations": null,
            "placeOfFinish": "Gyöngyöspata, általános iskola",
            "placeOfStart": "Gyöngyöspata, általános iskola",
            "price": 1600,
            "registerDeadline": "2018-07-21T00:00:00",
            "registrations": null,
          },
          Object {
            "beginningOfStart": "2018-07-22T06:00:00",
            "checkPoints": null,
            "distance": 34400,
            "elevation": 1338,
            "endOfStart": "2018-07-22T08:00:00",
            "hikeId": 2,
            "id": 3,
            "limitTime": "10:00:00",
            "maxNumOfHikers": 100,
            "name": "Patai Mátra 35",
            "participations": null,
            "placeOfFinish": "Gyöngyöspata, általános iskola",
            "placeOfStart": "Gyöngyöspata, általános iskola",
            "price": 1400,
            "registerDeadline": "2018-07-21T00:00:00",
            "registrations": null,
          },
        ],
        "date": "2018-07-22T00:00:00",
        "description": "Patai Mátra telejsítménytúra. Túra Gyöngyöspatáról 4 távon.",
        "id": 2,
        "name": "Patai Mátra",
        "organizer": null,
        "organizerId": 4,
        "staff": null,
        "website": "http://www.alfoldte.hu",
      }
    }
    key="2"
  />
  <HikeListItem
    hike={
      Object {
        "comments": null,
        "courses": Array [
          Object {
            "beginningOfStart": "2014-04-13T07:30:00",
            "checkPoints": null,
            "distance": 29750,
            "elevation": 915,
            "endOfStart": "2014-04-13T09:00:00",
            "hikeId": 3,
            "id": 4,
            "limitTime": "08:00:00",
            "maxNumOfHikers": 40,
            "name": "30 kilométeres táv.",
            "participations": null,
            "placeOfFinish": "Galgaguta, evangélikus templom melletti park",
            "placeOfStart": "Galgaguta, evangélikus templom melletti park",
            "price": 1200,
            "registerDeadline": "2014-04-10T00:00:00",
            "registrations": null,
          },
        ],
        "date": "2014-04-13T00:00:00",
        "description": "Andezit 30/15",
        "id": 3,
        "name": "Andezit",
        "organizer": null,
        "organizerId": 5,
        "staff": null,
        "website": "http://www.bthe.hu",
      }
    }
    key="3"
  />
</div>
`;

exports[`HikeListPage should render correctly while loading hikes 1`] = `
<p>
  Loading…
</p>
`;

exports[`HikeListPage should render correctly with errors 1`] = `
<p>
  Sorry! There was an error loading the items
</p>
`;
